app/lib/tariff_synchronizer/taric_update.rb

```diff
@@ -31,14 +45,33 @@ def correct_sequence_pair?(next_update, previous_update)
         next_sequence[:sequence].to_i == expected_next_sequence
       end
 
-      def download(date)
-        TaricUpdateDownloader.new(date).perform
-      end
-
       def update_type
         :taric
       end
 
+      def applicable_updates
+        # Pull out the most recent update
+        current_update = most_recent_pending || most_recent_applied
+
+        # Bail if there isn't one (impossible from where the data is now)
+        return [] if current_update.blank?
+        # Bail if we need to correct with a manual rollback
+        return [] unless correct_filename_sequence?
+
+        accumulator = []
```

Comment:
maybe rename this to `updates` or something less generic?

Reply:
This is a common patter for accumulating in iterators